frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts

```diff
@@ -26,93 +26,125 @@
 // See docs/COPYRIGHT.rdoc for more details.
 // ++
 
-import {Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {AfterViewInit, Component, ElementRef, EventEmitter, Input, OnDestroy, Output, ViewChild} from '@angular/core';
 import {ConfigurationService} from 'core-app/modules/common/config/configuration.service';
-import {DatePicker} from 'core-app/modules/common/op-date-picker/datepicker';
 import {TimezoneService} from 'core-components/datetime/timezone.service';
+import {DatePicker} from "core-app/modules/common/op-date-picker/datepicker";
+import {DebouncedEventEmitter} from "core-components/angular/debounced-event-emitter";
+import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
+import {componentDestroyed} from "@w11k/ngx-componentdestroyed";
+import {keyCodes} from "core-app/modules/common/keyCodes.enum";
+import {Instance} from "flatpickr/dist/types/instance";
 
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/7e7ade84990010b82932b3020f0c44e2101fe6a3/frontend/src/app/modules/common/op-date-picker/op-date-picker.component.ts#L38"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>